graphula-core/src/Graphula.hs

```diff
@@ -193,14 +201,36 @@ instance (MonadIO m, Applicative n, MonadIO n) => MonadGraphulaFrontend (Graphul
         Just key -> do
           existingKey <- get key
           whenNothing existingKey $ do
-            existingUnique <- checkUnique n
+            let updated = embedKey key n
```

Comment:
We could also do this out in `attemptsToInsertWith`. I think in either case, we'll end up logging the unchanged node, so we might wanna rejigger when we log.